releasenotes/notes/26185.yaml

```diff
@@ -0,0 +1,7 @@
+apiVersion: release-notes/v2
+kind: feature
+
+area: security
+
+releaseNotes: |
+  Add supports for the multi clusters with different trust domains, so mTLS would work in that case.
```

Comment:
probably make it more specific that this PR will use the trust domain of the remote cluster when generating the secure naming verify SAN.

Reply:
Why is mTLS not working though? is it an issue on the client side when it tries to validate the SANs of the remote k8s service [assuming the flatnetwork model, where cross cluster dns resolution is possible through some hacks]?

Reply:
Reworded the release notes. This PR include two things, support different trust domain when having different clusters, as well as another issue: even if trust domain is the same, httpbin is deployed in cluster1 with sa1, cluster2 with sa2. the secure naming should include both sa1 and sa2, but prior this PR, the serivce account only includes one sa with first found in the multi cluster reigstry, which makes 50% request fail.